### PR TITLE
Refs #32819 -- Established relationship between form fieldsets and their help text.

### PR DESCRIPTION
**Pull request by smithdc1**
[Ticket #32819](https://code.djangoproject.com/ticket/32819)

Added aria-described by for widgets rendered in a fieldset such as radios. aria-descrbiedby for these widgets is added to the `<fieldset>` element rather than each `<input>`.

### CODE REVIEW

File: django/forms/jinja2/django/forms/field.html [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {% if field.use_fieldset %}
-  <fieldset>
+  <fieldset{% if field.help_text and field.auto_id and "aria-describedby" not in field.field.widget.attrs %} aria-describedby="{{ field.auto_id }}_helptext"{% endif %}>
   {% if field.label %}{{ field.legend_tag() }}{% endif %}
 {% else %}
   {% if field.label %}{{ field.label_tag() }}{% endif %}
 {% endif %}
-{% if field.help_text %}<div class="helptext"{% if field.id_for_label %} id="{{ field.id_for_label}}_helptext"{% endif %}>{{ field.help_text|safe }}</div>{% endif %}
+{% if field.help_text %}<div class="helptext"{% if field.auto_id %} id="{{ field.auto_id }}_helptext"{% endif %}>{{ field.help_text|safe }}</div>{% endif %}
```
Comment (knyghty) [inline django/forms/jinja2/django/forms/field.html:7]: I like the cleaning up of the whitespace in here 💚

File: docs/ref/forms/fields.txt [hunk 2]
```diff
@@ -283,23 +283,33 @@ fields. We've specified ``auto_id=False`` to simplify the output:
...
 ``aria-describedby`` HTML attribute:
 
 .. code-block:: pycon
 
     >>> from django import forms
     >>> class UserForm(forms.Form):
     ...     username = forms.CharField(max_length=255, help_text="e.g., user@example.com")
-    ...
+    ...     color = forms.MultipleChoiceField(
+    ...         choices={1: "Red", 2:"Blue"},
```
Comment (knyghty) [inline docs/ref/forms/fields.txt:295]:
```suggestion
    ...         choices={1: "Red", 2: "Blue"},
```

**Review by knyghty — COMMENTED** [review]
I just wanted to come and say this is based on some testing we did and the approach seems good to me.

**Review by emarhin — APPROVED** [review]
Look clean and understand code for me and it works fine

File: django/forms/jinja2/django/forms/field.html (hunk 1 above)
Comment (felixxm) [inline django/forms/jinja2/django/forms/field.html:7]: Why did we globally change `id_for_label` to `auto_id`? :thinking: It looks like an unrelated cleanup (or even if it's related, it's something that can probably be processed separately).

Comment (smithdc1) [reply django/forms/jinja2/django/forms/field.html:7]: Sure. I'll pull this out into a separate PR and explain the reason for this change.

Comment (smithdc1) [reply django/forms/jinja2/django/forms/field.html:7]: Refs https://github.com/django/django/pull/17480

Comment (felixxm) [reply django/forms/jinja2/django/forms/field.html:7]: Thanks, please rebase.

**Review by felixxm — COMMENTED** [review]
@smithdc1 Thanks :+1:

I moved an extra test to the separate commit and pushed tiny edits.

File: tests/forms_tests/tests/test_forms.py [hunk 3]
```diff
@@ -3097,6 +3097,101 @@ class UserRegistration(Form):
...
+            )
+            radio = MultipleChoiceField(
+                choices=[("a", "A"), ("b", "B")],
+                widget=RadioSelect,
+                help_text="Radio help text",
+            )
+            datetime = SplitDateTimeField(help_text="Enter Date and Time")
+
+        f = FieldsetForm()
+        self.maxDiff = None
```
Comment (smithdc1) [inline tests/forms_tests/tests/test_forms.py:3132]: Need to chop this. Sorry.
